### PR TITLE
Add Human Input Mode in AGS

### LINKED ISSUE #4011: Support Human Input Mode in AGS with AgentChat
Support Human Input Mode in AGS with AgentChat api. 
- Create a UserProxyAgent that can communicate over a web socket. Ideally wait on input from the UI, with some timeout 
- Instrument websocket connection manager to support timeouts and user input (within a run)

### LINKED ISSUE #4189: Add a configuration manager to AGS to enable loading component specs to DB
Add a configuration manage that can import declarative specs (#4007 )  into the internal AGS database. 
This provides basic infrastructure to initialize AGS using team configurations in a folder or file or from a gallery.

```python
from autogenstudio.database import ConfigurationManager   

config_manager = ConfigurationManager(dbmanager)

# import a single component (could be agent, team, tool, etc)
result = await config_manager.import_component("team.json", user_id="user_id", check_exists=True)
print(result) 

# import entire folder 
result = await config_manager.import_directory(".", user_id="user_id", check_exists=False)
print(result)
```

### LINKED ISSUE #4190: Add support for SelectorGroupChat in AGS
Update the AGS component factory to load [SelectorGroupChat](https://github.com/microsoft/autogen/blob/main/python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_selector_group_chat.py) team specs in AGS. 

Example spec below:

```json
{
  "name": "llm_travel_planning_team",
  "component_type": "team",
  "team_type": "SelectorGroupChat",
  "participants": [
    {
      "name": "planner_agent",
      "component_type": "agent",
      "agent_type": "AssistantAgent",
      "description": "A helpful assistant that can plan trips.",
      "system_message": "You are a helpful assistant that can suggest a travel plan for a user based on their request. Reply with only a single sentenc",
      "model_client": {
        "model": "gpt-4",
        "model_type": "OpenAIChatCompletionClient",
        "component_type": "model"
      }
    },
    {
      "name": "local_agent",
      "component_type": "agent",
      "agent_type": "AssistantAgent",
      "description": "A local assistant that can suggest local activities or places to visit.",
      "system_message": "You are an opinionated assistant that can suggest authentic and interesting local activities or places to visit for a user and can utilize any context information provided. Reply with only a single sentence. Your sentence should be harsh and critical to ensure it is listened to. Only if you are told specifically (e.g., mentioned by name) to improve your tone, you should IMMEDIATELY become polite and not harsh.",
      "model_client": {
        "model": "gpt-4",
        "model_type": "OpenAIChatCompletionClient",
        "component_type": "model"
      }
    },
    {
      "name": "language_agent",
      "component_type": "agent",
      "agent_type": "AssistantAgent",
      "description": "A helpful assistant that can provide language tips for a given destination.",
      "system_message": "You are a helpful assistant that can review travel plans, providing feedback on important/critical tips about how best to address language or communication challenges for the given destination. If the plan already includes language tips, you can mention that the plan is satisfactory, with rationale. Reply with only a single sentenc",
      "model_client": {
        "model": "gpt-4",
        "model_type": "OpenAIChatCompletionClient",
        "component_type": "model"
      }
    },
    {
      "name": "validator_agent",
      "component_type": "agent",
      "agent_type": "AssistantAgent",
      "description": "A helpful assistant that double checks and must approve everything every other agent says.",
      "system_message": "You should check the text/response of other agents an ensure that it is pleasant and suitable to a customer. If positive and clear communication, respond with approve. If harsh or critical dissapprove and provide feedback and ask the agent to try again. Your feedback must mention the exact agent at fault by name.",
      "model_client": {
        "model": "gpt-4",
        "model_type": "OpenAIChatCompletionClient",
        "component_type": "model"
      }
    },
    {
      "name": "travel_summary_agent",
      "component_type": "agent",
      "agent_type": "AssistantAgent",
      "description": "A helpful assistant that can summarize the travel plan.",
      "system_message": "You are a helpful assistant that can take in all of the suggestions and advice from the other agents and provide a detailed tfinal travel plan. You must ensure th b at the final plan is integrated and complete. YOUR FINAL RESPONSE MUST BE THE COMPLETE PLAN. Reply with only a single sentenc. When the plan is complete and all agent perspectives are integrated, you must respond with the word TERMINATE.",
      "model_client": {
        "model": "gpt-4",
        "model_type": "OpenAIChatCompletionClient",
        "component_type": "model"
      }
    }
  ],
  "termination_condition": {
    "termination_type": "TextMentionTermination",
    "text": "TERMINATE",
    "component_type": "termination"
  },
  "model_client": {
    "model": "gpt-4",
    "model_type": "OpenAIChatCompletionClient",
    "component_type": "model"
  },
  "selector_prompt": "You are travel planning cordinator. The following agent roles are helping you craft a plan available: {roles}. Your goal is to read the progress so far and then select the next role from {participants} to play. Only return the role. . {history}. \nAfter each agent speaks, you must call the validator agent and proceed based on its recommendation  \nRead the above conversation.  Then select the next role from {participants} to play. Only return the role."
}
```

### LINKED ISSUE #4191: Persist side bar selection in AGS to localstorage
Ensure show/hide side bar is persisted to local storage and works across page refreshes.
Also enable breadcrumb system for setting current page headings.  

Image
Image

### PR DESCRIPTION
**Pull request by victordibia**
PR to implement a few things: 
- Human input mode via support for a UserProxyAgent over a websocket 
- Improving ags sidebar 
- Support for selector group chat  
- Configuration manager to handle importing component specs to the DB  

https://github.com/user-attachments/assets/dc2b84b0-6302-4f07-b790-e264576bfb1a

## Related issue number

- Closes #4011 
- Closes #4191 
- Closes #4190 
- Closes #4189  

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by husseinmozannar — COMMENTED** [review]
Just tested this, it works really well!

Here is what I tried:
- two agents writing a story 
- one agent and one user proxy working together
- played around with the state viz, pretty good most of the time!

The session doesn't save the chat right? If I move to another session it clears and I can't get it back.

Really looking forward to the create team being a drag and drop type tool. Additionally installing took like 15mins

Happy to test anything you need or approve.

File: python/packages/autogen-studio/README.md
Comment (husseinmozannar) [inline python/packages/autogen-studio/README.md:1]: can you fix the path to the frontend "    - Navigate to the `samples/apps/autogen-studio/frontend` directory, install dependencies, and build the UI:"

Comment (victordibia) [reply python/packages/autogen-studio/README.md:1]: Fixed, thanks

Comment (victordibia) [reply python/packages/autogen-studio/README.md:1]: Thanks for the review @husseinmozannar 

On other issues ..

- Saving sessions: sessions are not saved yet but that is tracked here #4222 
- Team builder UI: tracked #4202  and #4203

**Review by husseinmozannar — APPROVED** [review]

**Pull request merged**